plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -68,23 +68,21 @@ def _serialiseSettings(self, settings):
         prefix = ";SETTING_" + str(GCodeWriter.version) + " "  # The prefix to put before each line.
         prefix_length = len(prefix)
 
-        all_settings = InstanceContainer("G-code-imported-profile") #Create a new 'profile' with ALL settings so that the slice can be precisely reproduced.
-        all_settings.setDefinition(settings.getBottom())
-        for key in settings.getAllKeys():
-            all_settings.setProperty(key, "value", settings.getProperty(key, "value")) #Just copy everything over to the setting instance.
-        serialised = all_settings.serialize()
+        global_stack = Application.getInstance().getGlobalContainerStack()
+        container_with_profile = global_stack.findContainer({"type": "quality"})
+        serialized = container_with_profile.serialize()
```

Comment:
No, the new issue needs to be added. Didn't opened it so far.
Am 05.07.2016 22:20 schrieb "Aldo Hoeben" <notifications@github.com>:

> In plugins/GCodeWriter/GCodeWriter.py
> <https://github.com/Ultimaker/Cura/pull/821#discussion_r69630547>:
>
> > @@ -68,23 +68,21 @@ def _serialiseSettings(self, settings):
> >          prefix = ";SETTING_" + str(GCodeWriter.version) + " "  # The prefix to put before each line.
> >          prefix_length = len(prefix)
> >
> > -        all_settings = InstanceContainer("G-code-imported-profile") #Create a new 'profile' with ALL settings so that the slice can be precisely reproduced.
> > -        all_settings.setDefinition(settings.getBottom())
> > -        for key in settings.getAllKeys():
> > -            all_settings.setProperty(key, "value", settings.getProperty(key, "value")) #Just copy everything over to the setting instance.
> > -        serialised = all_settings.serialize()
> > +        global_stack = Application.getInstance().getGlobalContainerStack()
> > +        container_with_profile = global_stack.findContainer({"type": "quality"})
> > +        serialized = container_with_profile.serialize()
>
> In that case, consider this part reviewed and tested.
> Has the CURA-1615 ticket been updated with a link to the new issue?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/821/files/93cdce7b3360b2b285ca32f8b7cf7ba7522381a9#r69630547>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABwwjXkV6O2wEcGBdeAXy07TKfLmp0wpks5qSryXgaJpZM4I9EIA>
> .
>